tests/unit/neuralnets/NeuralLayerTestFixture.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2014, Shogun Toolbox Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ * contributors may be used to endorse or promote products derived from this
+ * software without specific prior written permission.
+
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Written (W) 2018 Saatvik Shah
+ */
+
+#pragma once
+
+#include "shogun/lib/SGMatrix.h"
+#include "shogun/lib/SGVector.h"
+#include "shogun/mathematics/Math.h"
+#include "shogun/neuralnets/NeuralInputLayer.h"
+#include "shogun/neuralnets/NeuralLinearLayer.h"
+
+#include "gtest/gtest.h"
+
+#include <memory>
+#include <tuple>
+
+using namespace shogun;
+
+class NeuralLayerTestFixture : public ::testing::Test
+{
+public:
+	template <typename T>
+	SGMatrix<T> create_rand_sgmat(
+	    int32_t num_rows, int32_t num_cols, T lower_bound, T upper_bound) const
+	{
+		auto data_v = SGVector<T>(num_rows * num_cols);
+		data_v.random(lower_bound, upper_bound);
```

Comment:
e.g. we could have a random method in SGMatrix (because we have one in vector already)
this code is a bit too much noise for my taste, especially in a test